packages/manager/apps/container/src/payment-modal/PaymentModal.tsx

```diff
@@ -0,0 +1,112 @@
+import React, { useEffect, useState } from 'react';
+import { Modal } from 'react-bootstrap';
+import { useTranslation } from 'react-i18next';
+
+import { useShell } from '@/context';
+
+import { fetchIcebergV6 } from '@ovh-ux/manager-core-api';
+import { useQuery } from '@tanstack/react-query';
+
+import { PAYMENT_ALERTS } from './constants';
+
+import './styles.scss';
+
+interface IPaymentMethod {
+  icon?: any;
+  label: string;
+  status: string;
+  default: boolean;
+  oneclick: boolean;
+  lastUpdate: string;
+  description: string;
+  integration: string;
+  paymentType: string;
+  creationDate: string;
+  paymentMeanId: number;
+  expirationDate?: string;
+  paymentSubType: string;
+  paymentMethodId: number;
+}
+
+const PaymentModal = (): JSX.Element => {
+  const [alert, setAlert] = useState('');
+  const { t } = useTranslation('payment-modal');
+  const [showPaymentModal, setShowPaymentModal] = useState(false);
+  const shell = useShell();
+
+  const paymentMethodURL = shell
+    .getPlugin('navigation')
+    .getURL('dedicated', '#/billing/payment/method');
+
+  const closeHandler = () => setShowPaymentModal(false);
+
+  const computeAlert = (paymentMethods: IPaymentMethod[]): string => {
+    let alert: string = null
+    const hasDefaultPaymentMethod: IPaymentMethod = paymentMethods?.find(currentPaymentMethod => currentPaymentMethod.default);
+    if (!hasDefaultPaymentMethod) {
+      alert = PAYMENT_ALERTS.NO_DEFAULT;
+    } else {
+      const currentCreditCard: IPaymentMethod = paymentMethods?.find(currentPaymentMethod => currentPaymentMethod.paymentType === 'CREDIT_CARD');
+      if (currentCreditCard) {
+        const creditCardExpirationDate: Date = new Date(currentCreditCard.expirationDate);
+        if (creditCardExpirationDate.getTime() < Date.now()) {
+          alert = PAYMENT_ALERTS.EXPIRED_CARD;
+        } else {
+          const currentDateMinus30Days: Date = new Date();
+          currentDateMinus30Days.setDate(currentDateMinus30Days.getDate() - 30);
+          const isSoonToBeExpireCreditCard: boolean = currentDateMinus30Days.getTime() > Date.now() && creditCardExpirationDate.getTime() > Date.now();
+          if (isSoonToBeExpireCreditCard) {
+            alert = PAYMENT_ALERTS.SOON_EXPIRED_CARD;
+          }
+        }
+      }
+    }
+    return alert;
+  };
```

Comment:
Can you move this function outside the component for better readability ?

Reply:
done